corehq/motech/value_source.py

```diff
@@ -143,7 +145,71 @@ def check_direction(self, direction):
         return not self.direction or direction == self.direction
 
 
-class CaseProperty(ValueSource):
+class MapMixin(DocumentSchema):
```

Comment:
Does this need to inherit from `DocumentSchema`? That's dragging along quite a bit of baggage, and I'm assuming all classes that use `MapMixin` will already have `DocumentSchema` in their inheritance hierarchy via some other base class.

Reply:
If you don't extend anything, `jsonobject` gets upset:
```
Traceback (most recent call last):
  File "corehq/motech/tests/test_value_source.py", line 328, in test_no_values
    external_value = value_source.get_import_value(json_doc)
  File "corehq/motech/value_source.py", line 209, in get_import_value
    return self.deserialize(external_value)
  File "corehq/motech/value_source.py", line 182, in deserialize
    if self.value_map is None:
  File "jsonobject/base_properties.pyx", line 73, in jsonobject.base_properties.JsonProperty.__get__
AssertionError
```

But you make a good point. We can avoid the baggage added by `DocumentSchema` if we just extend `JsonObject`. I noticed [some UCR classes](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/expressions/specs.py#L39) do this, and maybe this is why.

Reply:
oof. the MRO for this class is even more complex than I realized